apps/emqx_connector_aggregator/src/emqx_connector_aggregator.erl

```diff
@@ -99,10 +99,12 @@ write_records_limited(Name, Buffer = #buffer{max_records = MaxRecords}, Records)
             write_records(Name, Buffer, Records)
     end.
 
-write_records(Name, Buffer = #buffer{fd = Writer}, Records) ->
+write_records(Name, Buffer = #buffer{fd = Writer, max_records = MaxRecords}, Records) ->
     case emqx_connector_aggreg_buffer:write(Records, Writer) of
         ok ->
             ?tp(connector_aggreg_records_written, #{action => Name, records => Records}),
+            get_num_records(Buffer) >= MaxRecords andalso
```

Comment:
why not line 95?

Reply:
data has not been written at that point.  there, the buffer is rotated prior to actually writing, we want to write and then trigger the rotation if max has been reached.